### PR TITLE
Add routers for login, home, trip detail pages

### PR DESCRIPTION
**Pull request by yongbingao**
@kevinlin505 @wleofeng check it out

### CODE REVIEW

File: src/components/SignIn/SignIn.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const SignIn = ({ actions, auth }) => {
 
   return (
     <div>
+      <h2>GoPlan</h2>
       {auth.isAuthenticated ? (
         <button onClick={actions.auth.signOut}>Sign Out</button>
```
Comment (kevinlin505) [inline src/components/SignIn/SignIn.js]: I don't think we need this anymore right?

File: src/providers/auth/auth.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ export default function reducer(state = initialState, action) {
 
 export const authActions = {
   checkAuth: () => dispatch => {
-    initializeFireBase();
+    if(!firebase.apps.length) initializeFireBase();
```
Comment (kevinlin505) [inline src/providers/auth/auth.js]: I already included this in the `_db.js` file.

File: src/utils/router_util.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Route, Redirect, withRouter } from 'react-router-dom';
+
+const msp = state => {
+  return { auth: Boolean(state.auth.isAuthenticated && state.auth.profile) };
```
Comment (kevinlin505) [inline src/utils/router_util.js]: you can just do `!!(state.auth.isAuthenticated && state.auth.profile)`

File: src/utils/router_util.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Route, Redirect, withRouter } from 'react-router-dom';
+
+const msp = state => {
```
Comment (kevinlin505) [inline src/utils/router_util.js]: Also, can we do something like `mapStateToProps` and `mapDispatchToProps` instead? Always good to be descriptive on names.

File: src/components/SignIn/SignIn.js (hunk 1 above)
Comment (yongbingao) [reply src/components/SignIn/SignIn.js]: I removed the HomePage file, so theres no title right now on the login page.

Comment (kevinlin505) [reply src/components/SignIn/SignIn.js]: I mean the sign out button and `auth.isAuthenticated ?`

File: src/utils/router_util.js (hunk 4 above)
Comment (yongbingao) [reply src/utils/router_util.js]: made descriptive names

**Comment by yongbingao** [comment]
@kevinlin505 take a look at the comment resolutions

File: src/components/UserPage/UserPage.js [hunk 5]
```diff
@@ -20,7 +20,9 @@ const UserPage = ({ actions, auth, trip }) => {
   }, [auth.profile.trips.length]);
 
   useEffect(() => {
+    debugger;
     const trips = trip.trips.map((tripDetails, index) => {
+      debugger;
```
Comment (kevinlin505) [inline src/components/UserPage/UserPage.js]: remove these debuggers

Comment (yongbingao) [reply src/components/UserPage/UserPage.js]: my bad, forgot about those

**Pull request merged**